src/util/StringUtils.h

```diff
@@ -246,8 +246,8 @@ std::string insertThousandSeparator(const std::string_view str,
     constexpr ctll::fixed_string symbolAsFixedString({symbol, '\0'});
 
     // Inside a regex character class are fewer reserved character.
-    if constexpr (symbol == '^' || symbol == '-' || symbol == ']' ||
-                  symbol == '\\') {
+    if constexpr (symbol == '^' || symbol == '-' || symbol == '[' ||
```

Comment:
I think you can make this a normal function with the symbol as an argument, and then use
`ad_utility::contains(string_view{"^-{}\\"}, symbol)` (after adding `constexpr` to tha `contains` algorithm.
Lambdas are implicitly contexpr, so you can just use
`constexpr fixed_string bla = adjustSymbolForRegex(signifier);`

which is shorter and easier to read.

Reply:
As far as I can tell, the usage of `util/Algorithm.h` inside `util/StringUtil.h` is currently not possible because of a cyclic `#include` chain over multiple files.

Reply:
While the wanted change wasn't possible, using a helper function in `std::string_view` to reduce the number of calls was. See [bbb9aad](https://github.com/ad-freiburg/qlever/pull/1182/commits/bbb9aadba8f7d46beeb80b318a54b4c7b637bf29).

Reply:
Okay, that is a pity, but your manual way with find and npos is also okay:)

Reply:
Alright, good news: The master was recently updated and not it works! (See [384e409](https://github.com/ad-freiburg/qlever/pull/1182/commits/384e40941496c46071b42166d645364175f6ba8e).)

However, I've had to change a few things in `ad_utility::contains`, so that everything worked cleaner.